docs/basic_concepts.rst

```diff
@@ -0,0 +1,172 @@
+parameters specify which address will be generated from it. The process is
+deterministic, meaning that same input paramteres always generate the same address.
+
+Addresses are 81 trytes long and may contain extra 9 trytes for checksum.
+The checksum may be used to verify that an address is in fact a valid IOTA address.
+
+For PyOTA specific implementation details on addresses, see :py:class:`Address`.
+
+Transaction
+-----------
+
+From the `official IOTA documentation site`_:
+
+*A transaction is a single transfer instruction that can either withdraw IOTA
+tokens from an address, deposit them into an address, or have zero-value
+(contain data, a message, or a signature). If you want to send anything to an
+IOTA network, you must send it to a node as a transaction.*
+
+Transactions are always 2673 trytes long and their `structure`_ is defined by
+the protocol. They can be classified into three categories:
+
+- `Input transaction`_: A transaction that withdraws tokens from an address.
+- `Output transaction`_: A transaction that deposits tokens to an address.
+- `Zero-value transaction`_: A transaction that has 0 value and might carry messages
+  or signatures.
+
+Depending on the type of the transaction, different fields are required to be filled.
+
+A transaction's unique identifier in IOTA is the :py:class:`TransactionHash`,
+that is generated from the trytes of the transaction. If any trytes change
+in the transaction, the returning transaction hash would alter. This way, transaction
+hashes ensure the immutability of the Tangle.
+
+To become accepted by the network, a transaction has to be attached to the Tangle.
+The attachment process means that the transaction should reference two unconfirmed
+transactions (tips) in the Tangle and do a small proof-of-work. This process might
+be performed by a node, or by using the local proof-of-work feature of the `client
+libraries`_.
+
+For PyOTA specific implementation details on transactions, see :py:class:`Transaction`
+and :py:class:`ProposedTransaction`.
+
+Bundle
+------
+
+From the `official IOTA documentation site`_:
+
+*A bundle is a group of transactions that rely on each other's validity.
+For example, a transaction that deposits IOTA tokens into an address relies on
+another transaction to withdraw those IOTA tokens from another address.
+Therefore, those transactions must be in the same bundle.*
+
+In other words, a bundle is collection of transactions, treated as an atomic unit when
+attached to the Tangle.
+
+.. note::
+    Unlike a block in a blockchain, bundles are not first-class
+    citizens in IOTA; only transactions get stored in the Tangle.
+
+    Instead, bundles must be inferred by following linked transactions
+    with the same bundle hash.
+
+Transactions in the bundle are linked together through their ``branchTransaction``
+and ``trunkTransaction`` fields, furthermore they are indexed within the bundle
```

Comment:
❕ I believe the client libraries rely exclusively on `trunkTransaction` to link bundle transactions together; as far as I'm aware, `branchTransaction` does not have any function beyond confirming a second tip.

Reply:
You are right. What do you think about including [such](https://docs.iota.org/assets/docs/getting-started/0.1/images/bundle-structure.png) a picture to illustrate this?

Reply:
Ooh, ya, that looks good!